File: ruby/gigasecond/gigasecond.rb
class Gigasecond
  
  GIGASECOND = 1_000_000_000
  DAY_IN_SECONDS = 60 * 60 * 24
  GIGASECOND_TO_DAYS = GIGASECOND / DAY_IN_SECONDS

  def initialize(start_date)
    @start_date = start_date
  end

  def date
    @start_date + GIGASECOND_TO_DAYS
  end

end

File: javascript/gigasecond/gigasecond.js
(function() {

  'use strict';

  function Gigasecond(birthDate) {
    this.birthDate = birthDate;
    this.interval = 1000000000000;
  };

  Gigasecond.prototype.date = function() {
    var gigasecondCelebrationDate = new Date(this.birthDate.getTime() + this.interval);
    return this._beginningOfTheDay(gigasecondCelebrationDate);
  };

  Gigasecond.prototype._beginningOfTheDay = function(date) {
    date.setSeconds(0);
    date.setMinutes(0);
    date.setHours(0);
    return date;
  };

  module.exports = Gigasecond;

})();